src/handlers/internal-server-error-handler.ts

```diff
@@ -1,36 +1,52 @@
 import { HTTP_STATUS_CODES } from "../app.constants";
-import axios from "axios";
+import { isAxiosError } from "axios";
 import PAGES from "../constants/ipv-pages";
 import { getIpvPageTemplatePath, getErrorPageTemplatePath } from "../lib/paths";
 import ERROR_PAGES from "../constants/error-pages";
 import { ErrorRequestHandler } from "express";
+import HttpError from "../errors/http-error";
 
-const hasStatus = (input: unknown): input is { status: number } =>
-  !!(input as { status: number }).status;
+export const HANDLED_ERROR = new Error("Placeholder error for events logged in error handler");
+
+const getErrorStatus = (err: unknown): number => {
+  if (isAxiosError(err)) {
+    if (err.response?.status) {
+      return err.response.status;
+    }
+  } else if (err instanceof HttpError) {
+    return err.status;
+  }
+  return HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR;
+};
 
 const serverErrorHandler: ErrorRequestHandler = (err, req, res, next) => {
   if (res.headersSent) {
     return next(err);
   }
 
-  if (!axios.isAxiosError(err)) {
-    req.log.error({
-      message: { err, message: "An internal server error occured" },
-      level: "ERROR",
-    });
-  }
+  const status = getErrorStatus(err);
+
+  req.log?.error({
+    message: {
+      description: err?.constructor?.name ?? "Unknown error",
+      errorMessage: err?.message ?? "Unkown error",
+      errorStack: err?.stack,
+    },
+  });
+
+  res.status(status);
 
   if (res.statusCode === HTTP_STATUS_CODES.UNAUTHORIZED) {
     return res.render(getErrorPageTemplatePath(ERROR_PAGES.SESSION_ENDED));
   }
-  res.err = err; // this is required so that the pino logger does not log new error with a different stack trace
 
-  if (hasStatus(res.err)) {
-    res.status(res.err.status);
-  } else {
-    res.status(HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR);
+  if (res.statusCode === HTTP_STATUS_CODES.NOT_FOUND) {
+    return res.render(getErrorPageTemplatePath(ERROR_PAGES.PAGE_NOT_FOUND));
   }
 
+  // Set this to avoid pino-http generating a new error in the request log
+  res.err = HANDLED_ERROR;
```

Comment:
This is a bit awkward, but I didn't really want the error to be logged again by the pino-http handler...